chore: update changelog for 2.0.0 rc release (#1208)
juju/terraform-provider-juju: CHANGELOG.md

# 2.0.0

NOTES:
* * **This release requires Juju controller version 3 or higher. This version drops the support for 2.9**
* **If using JAAS, this release requires Juju controller version 3.6.5 or higher.**
* This release uses Juju client api code from the tip of the Juju 4.0 branch.

ENHANCEMENTS

* Enable the Terraform provider to run against Juju 3 & 4 controllers.

> * If you are running against a Juju 3 controller, upgrading from latest v1 of the Terraform Provider to v2 should require no changes. See the [upgrade to v2 guide](https://documentation.ubuntu.com/terraform-provider-juju/latest/howto/manage-provider/upgrade-provider-to-v2/) for more information.

Reviewer: ```suggestion
* If you are running against a Juju 3 controller, upgrading from the latest v1 of the Terraform Provider to v2 should require no changes. See the [upgrade to v2 guide](https://documentation.ubuntu.com/terraform-provider-juju/latest/howto/manage-provider/upgrade-provider-to-v2/) for more information.
```

# 2.0.0

NOTES:
* * **This release requires Juju controller version 3 or higher. This version drops the support for 2.9**
* **If using JAAS, this release requires Juju controller version 3.6.5 or higher.**
* This release uses Juju client api code from the tip of the Juju 4.0 branch.

ENHANCEMENTS

* Enable the Terraform provider to run against Juju 3 & 4 controllers.

> * If you were using a v1 of the Terraform Provider and a Juju 3 controller, if you upgrade to v2 

Reviewer: I think we need a small doc on upgrading from v1 of the provider to v2 because you technically need to be on the latest v1, you can't just upgrade from any old version. Then we can link the doc here. 

In addition, I think we can do some rewording,
```suggestion
* Upgrading from v1 of the Terraform Provider to v2 should require no changes to your plan or infra.
```

Author: a doc? can we just say from latest v1 to v2?

Author: something like:
`* Upgrading from latest v1 of the Terraform Provider to v2 should require no changes to your plan or infra.`

Reviewer: I think we should have a doc just like we have https://documentation.ubuntu.com/terraform-provider-juju/v1/howto/manage-provider/upgrade-provider-to-v1/. It's a whole major version upgrade, and the only place we mention upgrading shouldn't be in our changelog imo.

Reviewer: It'll be very short, but it will make searching for the info easier.

Author: ok, i'll add it in this pr then

Author: added

  it should require no changes.

BREAKING CHANGES

> * Drop support for 2.9 controllers. We will keep supporting 2.9 controllers in the v1 track.

Reviewer: Would it be more accurate to say,
```suggestion
* Drop support for 2.9 controllers. We will keep supporting 2.9 controllers in the v1 track for security updates and bug fixes.
```
